sympy/printing/codeprinter.py

```diff
@@ -67,6 +69,15 @@ def __init__(self, settings=None):
         if not hasattr(self, 'reserved_words'):
             self.reserved_words = set()
 
+    def _handle_UnevaluatedExpr(self, expr):
+        if expr is None:
+            return expr
+        elif isinstance(expr, Basic):
+            return expr.replace(re, lambda arg: arg if isinstance(
+                arg, UnevaluatedExpr) and arg.args[0].is_real else re(arg))
+        else:
+            return type(expr)(self._handle_UnevaluatedExpr(item) for item in expr)
```

Comment:
Why does this need to handle non-Basic?

Reply:
I just pushed a commit which moves this non-Basic handling into lambdify's internal class.